### PR TITLE
Fix OTA FreeRTOS port and other minor fixes in MQTT

### PR DESCRIPTION
**Pull request by pvyawaha**
*Issue #, if available:*

*Description of changes:*
Fix OTA FreeRTOS port 
Make topic buffers static 
Add OTA FreeRTOS port error codes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/include/ota.h [hunk 1]
```diff
@@ -323,6 +323,11 @@ typedef struct OtaAgentContext
 #define OTA_ERR_SUBSCRIBE_FAILED             0x40000000U /*!< Failed to subscribe to a topic. */
 #define OTA_ERR_UNSUBSCRIBE_FAILED           0x41000000U /*!< Failed to unsubscribe from a topic. */
 #define OTA_ERR_FAILED_TO_DECODE_CBOR        0x42000000U /*!< Failed to decode CBOR object. */
+#define OTA_ERR_TIMER_CREATE_FAILED          0x43000000U /*!< Failed to create timer. */
```
Comment (divekarshubham) [inline source/include/ota.h:326]: Why not use the same error codes as
https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/38f3af7d4e6219961fcacc0c2bc56eaea47252b8/source/include/ota.h#L319 ?

File: source/portable/os/ota_os_freertos.c [hunk 2]
```diff
@@ -312,10 +331,14 @@ OtaErr_t OtaStopTimer_FreeRTOS( OtaTimerId_t otaTimerId )
...
 {
+    configASSERT( ( otaTimerId >= OtaRequestTimer ) && ( otaTimerId < OtaNumOfTimers ) );
+
     OtaErr_t otaErrRet = OTA_ERR_UNINITIALIZED;
 
+    BaseType_t retVal = pdFALSE;
+
     if( otaTimer[ otaTimerId ] != NULL )
     {
         /* Stop the timer. */
```
Comment (divekarshubham) [inline source/portable/os/ota_os_freertos.c:344]: Should update the comment to say 'Delete the timer'

File: source/include/ota.h (hunk 1 above)
Comment (pvyawaha) [reply source/include/ota.h:326]: Yes seems a mistake

File: source/portable/os/ota_os_freertos.c (hunk 2 above)
Comment (pvyawaha) [reply source/portable/os/ota_os_freertos.c:344]: Thanks , updating it

Comment (divekarshubham) [reply source/portable/os/ota_os_freertos.c]: Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/91/commits/935a2ddcdf079e6aca118c6dd4266df35a825df0

File: source/include/ota.h (hunk 1 above)
Comment (divekarshubham) [reply source/include/ota.h]: Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/91/commits/935a2ddcdf079e6aca118c6dd4266df35a825df0

**Review by divekarshubham — APPROVED** [review]

File: source/portable/os/ota_os_freertos.c [hunk 3]
```diff
@@ -279,8 +294,12 @@ OtaErr_t OtaStartTimer_FreeRTOS( OtaTimerId_t otaTimerId,
 
 OtaErr_t OtaStopTimer_FreeRTOS( OtaTimerId_t otaTimerId )
 {
+    configASSERT( ( otaTimerId >= OtaRequestTimer ) && ( otaTimerId < OtaNumOfTimers ) );
```
Comment (aggarg) [inline source/portable/os/ota_os_freertos.c:297]: Same as above i.e.

```c
OtaErr_t otaErrRet = OTA_ERR_UNINITIALIZED;
BaseType_t retVal = pdFALSE;

configASSERT( ( otaTimerId >= OtaRequestTimer ) && ( otaTimerId < OtaNumOfTimers ) );
```

File: source/portable/os/ota_os_freertos.c [hunk 4]
```diff
@@ -155,21 +160,27 @@ OtaErr_t OtaReceiveEvent_FreeRTOS( OtaEventContext_t * pContext,
     return otaErrRet;
 }
 
-void OtaDeinitEvent_FreeRTOS( OtaEventContext_t * pContext )
+OtaErr_t OtaDeinitEvent_FreeRTOS( OtaEventContext_t * pContext )
 {
     ( void ) pContext;
 
+    OtaErr_t otaErrRet = OTA_ERR_NONE;
```
Comment (aggarg) [inline source/portable/os/ota_os_freertos.c:167]: I remember seeing build warning/failure on some platform if the first code statement is before all the local declarations. Consider moving the assert after all the local variables are declared i.e.
```c
OtaErr_t otaErrRet = OTA_ERR_NONE;

( void ) pContext;
```

Comment (divekarshubham) [reply source/portable/os/ota_os_freertos.c:163]: Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/91/commits/5b69254cbead6a8843cb66190090e39e9940e6ce

File: source/portable/os/ota_os_freertos.c (hunk 3 above)
Comment (divekarshubham) [reply source/portable/os/ota_os_freertos.c]: Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/commit/5b69254cbead6a8843cb66190090e39e9940e6ce

**Review by aggarg — APPROVED** [review]

**Review by divekarshubham — APPROVED** [review]

**Pull request merged**